### PR TITLE
Improve Content Security Policy documentation [ci-skip]

### PR DESCRIPTION
**Pull request by p8**
### Summary

- Use "HTTP Content-Security-Policy response header" instead of "Content
  Security Policy", to make it clear the header will be set.
- Document `report_uri` in the guides.
- Instead of having a long list of examples in the guide, add a
  description to each example.

### Before
image
image
image

### After
image
image
image

### CODE REVIEW

File: guides/source/security.md [hunk 1]
```diff
@@ -1118,34 +1115,49 @@ class PostsController < ApplicationController
...
-in order to report only content violations for migrating
-legacy content
+response header to only report content violations:
 
 ```ruby
 # config/initializers/content_security_policy.rb
 Rails.application.config.content_security_policy_report_only = true
-```
 
-```ruby
```
Comment (jonathanhefner) [inline guides/source/security.md:1148]: When code snippets are from different contexts (e.g. `config/initializers/content_security_policy.rb` and `app/controllers/posts_controller.rb`), I prefer to keep the code blocks separate to provide visual distinction in the rendered output.  Also, it gives each snippet a dedicated "Copy" button.

(I did some work toward that in #40922.)

**Review by jonathanhefner — COMMENTED** [review]
Nice! :+1:

File: guides/source/security.md [hunk 2]
```diff
@@ -1118,34 +1115,49 @@ class PostsController < ApplicationController
...
+[report-uri](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/report-uri)
+directive to report violation to the specified URI:
+
+```ruby
+Rails.application.config.content_security_policy do |policy|
+  policy.report_uri "/csp-violation-report-endpoint"
+end
+```
+
+When migrating legacy content set the
```
[the remark this answers is not in the document]
Comment (p8) [reply guides/source/security.md]: Thanks!

File: guides/source/security.md (hunk 1 above)
Comment (p8) [reply guides/source/security.md:1148]: Yes, good point. I agree.

File: guides/source/security.md [hunk 3]
```diff
@@ -1118,34 +1115,49 @@ class PostsController < ApplicationController
     policy.base_uri "https://www.example.com"
   end
 end
```
[the remark this answers is not in the document]
Comment (p8) [reply guides/source/security.md:1109]: Yes, that looks better.

**Pull request merged**